### PR TITLE
CXX-Qt-lib: Add QModelIndex::internalPointer

### LINKED ISSUE #719: QModelIndex::internalId() needed for trees
Possible tasks:

- [ ] Add Qt types to cxx-qt-lib, eg `quintptr` so it can be used easily in bridges https://doc.qt.io/qt-6/qttypes.html
- [ ] Add an alias in a C++ header somewhere for `c_void` as suggested in https://github.com/dtolnay/cxx/issues/1049
- [x] Add support for methods in existing types which return these types and were missed before, eg `QModelIndex::internalId`

Original description:

I am trying to create a tree-model to provide data to QML. 
The normal way is to use QAbstractItemModel, but there is a problem:
The view might request an items parent by calling QModelIndex::parent().

In a list or table, where items never have parents, this would simply return an invalid QModelIndex().
The current item can then be found simply by it's row and column.

In a tree, the parent is needed as well, for which I need to know the current Item, for which I need the parent ...
That problem is usually solved by storing a pointer or id in the QModelIndex, which is set by QAbstractItemModel::createIndex() and retrieved by QModelIndex::internalId() (or internalPointer()) -- but cxx_qt does not provide access to these methods.

I thought about a workaround by following the parents up to the root level, remembering all the rows and then down again to find the right item, but as the documentation of QAbstractItemModel::parent states:

> When reimplementing this function in a subclass, be careful to avoid calling QModelIndex member functions, such as QModelIndex::parent(), since indexes belonging to your model will simply call your implementation, leading to infinite recursion.

so that would not work.

The only other way I can think of, that might possibly work, is subclassing QModelIndex, which I'd like to avoid.

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
Closes #719 

Depends on #738

### CODE REVIEW

File: crates/cxx-qt-lib/src/core/qmodelindex.rs [hunk 1]
```diff
@@ -34,11 +34,21 @@ mod ffi {
...
         fn siblingAtRow(self: &QModelIndex, row: i32) -> QModelIndex;
+
+        /// Returns a `*mut c_void` pointer used by the model to associate the index with the internal data structure.
+        #[rust_name = "internal_pointer_mut"]
+        fn internalPointer(self: &QModelIndex) -> *mut c_void;
+
+        /// Returns a `*const c_void` pointer used by the model to associate the index with the
+        /// internal data structure.
+        #[rust_name = "internal_pointer"]
+        fn constInternalPointer(self: &QModelIndex) -> *const c_void;
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/core/qmodelindex.rs:45]: this only exists in Qt 6...

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-lib/src/core/qmodelindex.rs:45]: Yeah, CI noticed as well :see_no_evil: 

So I guess we should really try getting the `#[cfg()]` attributes working in CXX then I guess.
Has there been any progress on this for #746 ?

Comment (ahayzen-kdab) [reply crates/cxx-qt-lib/src/core/qmodelindex.rs:45]: Not yet, we need to see if it works in a mini example then try and get it working with cxx-qt-lib

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-lib/src/core/qmodelindex.rs]: I've removed the `const` version of this function for now.
It's not all that useful, as you can always cast to a `const*` from a `mut*` anyway.

We can always readd it if people ask for it.

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**